### PR TITLE
#SB-26905 feat: Transcripts for player (video/audio) [Content player] V2

### PR DESCRIPTION
**Pull request by santoshilimi**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-26905" title="SB-26905" target="_blank">StorySB-26905</a>  Transcripts for player (video/audio) [Content player] V2
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export class SunbirdVideoPlayerComponent implements OnInit, AfterViewInit, OnDes
   showQumlPlayer = false;
   QumlPlayerConfig: any = {};
   videoInstance: any;
+  transcripts: any;
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:38]: Keep proper type with interface

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: projects/sunbird-video-player/src/lib/components/video-player/video-player.component.html [hunk 2]
```diff
@@ -1,4 +1,6 @@
-<video #target class="video-js" controls (loadeddata)="onLoadMetadata($event)"></video>
+<video #target class="video-js" controls (loadeddata)="onLoadMetadata($event)">
+  <track *ngFor="let trans of transcripts" kind="captions" src="{{trans.artifactUrl}}" srclang="{{trans.language}}" label="{{trans.language}}" >
```
Comment (vinukumar-vs) [inline projects/sunbird-video-player/src/lib/components/video-player/video-player.component.html:2]: srclang="{{trans.language}}" use languageCode

File: projects/sunbird-video-player/src/lib/components/video-player/video-player.component.scss [hunk 3]
```diff
@@ -196,4 +196,8 @@
 
 div[data-marker-key] {
   margin-left: 0.7% !important;
+}
+
+.vjs-texttrack-settings {
+  display: none;
```
Comment (vinukumar-vs) [inline projects/sunbird-video-player/src/lib/components/video-player/video-player.component.scss:202]: Add the comments why are you disabled. Add TODO comment

File: src/app/app.component.ts [hunk 4]
```diff
@@ -45,7 +45,30 @@ export class AppComponent {
...
+      transcripts: [
+        {
+          language: 'Bengali',
+          identifier: 'do_11353887890259968011412',
```
Comment (vinukumar-vs) [inline src/app/app.component.ts:54]: Add "languageCode" for all the defined lanaguages.

Comment (vinukumar-vs) [reply src/app/app.component.ts:54]: You have to verify for all the languages..

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts [hunk 5]
```diff
@@ -35,6 +36,7 @@ export class SunbirdVideoPlayerComponent implements OnInit, AfterViewInit, OnDes
   showQumlPlayer = false;
   QumlPlayerConfig: any = {};
   videoInstance: any;
+  transcripts: Transcripts;
```
Comment (vinukumar-vs) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:39]: It's already there as part of this.playerConfig. why do you need one more vaiable here.
Remove it if you used only once or twice.

File: projects/sunbird-video-player/src/lib/playerInterfaces.ts [hunk 6]
```diff
@@ -69,3 +69,9 @@ export interface PlayerConfig {
     metadata: any; // content
     data?: any; // body
 }
+export interface Transcript {
+    language: string;
```
Comment (vinukumar-vs) [inline projects/sunbird-video-player/src/lib/playerInterfaces.ts:73]: LanguageCode need to add here

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.html [hunk 7]
```diff
@@ -2,7 +2,7 @@
       <sb-player-side-menu-icon class="sb-player-side-menu-icon notVisible" tabindex="0" (sidebarMenuEvent)="sideBarEvents($event)"
 
             *ngIf="viewState ==='player'" [ngClass]="{'isVisible': showControls}"></sb-player-side-menu-icon>
-      <video-player *ngIf="viewState === 'player'" (questionSetData)="questionSetData($event)" (playerInstance)="playerInstance($event)" [config]="playerConfig.config">
+      <video-player *ngIf="viewState === 'player'" (questionSetData)="questionSetData($event)" (playerInstance)="playerInstance($event)" [config]="playerConfig.config" [transcripts]="transcripts">
```
Comment (vinukumar-vs) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.component.html:5]: Check why are we passing this separatly.

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts [hunk 8]
```diff
@@ -5,6 +5,7 @@ import {
 import { ErrorService , errorCode , errorMessage, ISideBarEvent } from '@project-sunbird/sunbird-player-sdk-v9';
 
 import { PlayerConfig } from './playerInterfaces';
+import { Transcripts } from './playerInterfaces';
```
Comment (vinukumar-vs) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:8]: move this logic to ViewerService & pass it as part 'transcripts' as part of PlayerConfig.config itself.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: projects/sunbird-video-player/src/lib/components/video-player/video-player.component.scss (hunk 3 above)
Comment (santoshilimi) [reply projects/sunbird-video-player/src/lib/components/video-player/video-player.component.scss:202]: Done

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts (hunk 5 above)
Comment (santoshilimi) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts]: Done

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.html (hunk 7 above)
Comment (santoshilimi) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.html]: Done

File: projects/sunbird-video-player/src/lib/playerInterfaces.ts (hunk 6 above)
Comment (santoshilimi) [reply projects/sunbird-video-player/src/lib/playerInterfaces.ts:73]: Done

File: projects/sunbird-video-player/src/lib/services/viewer.service.ts [hunk 9]
```diff
@@ -80,7 +82,26 @@ export class ViewerService {
       this.mimeType = metadata.mimeType;
     }
   }
-
+  handleTranscriptsData(transcripts) {
+    if (_.isArray(transcripts)) {
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/services/viewer.service.ts:86]: if (!_.isArray(transcripts)) { 
this.raiseExceptionLog('INVALID_TRANSCRIPT_DATATYPE', 'TRANSCRIPT', new Error('Transcript data should be array'), this.traceId);
return [];
}

File: projects/sunbird-video-player/src/lib/services/viewer.service.ts [hunk 10]
```diff
@@ -221,10 +243,11 @@ export class ViewerService {
     const interactItems = ['PLAY', 'PAUSE', 'EXIT', 'VOLUME_CHANGE', 'DRAG',
       'RATE_CHANGE', 'CLOSE_DOWNLOAD', 'DOWNLOAD', 'NAVIGATE_TO_PAGE',
       'NEXT', 'OPEN_MENU', 'PREVIOUS', 'CLOSE_MENU', 'DOWNLOAD_MENU', 'DOWNLOAD_POPUP_CLOSE', 'DOWNLOAD_POPUP_CANCEL',
-     'SHARE', 'REPLAY', 'FORWARD', 'BACKWARD', 'FULLSCREEN' , 'NEXT_CONTENT_PLAY',
+      'SHARE', 'REPLAY', 'FORWARD', 'BACKWARD', 'FULLSCREEN', 'NEXT_CONTENT_PLAY', 'transcript_language_selected',
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/services/viewer.service.ts:246]: Follow standard

File: projects/sunbird-video-player/src/lib/sunbird-video-player.service.ts [hunk 11]
```diff
@@ -105,10 +105,10 @@ export class SunbirdVideoPlayerService {
 
   }
 
-  public interact(id, currentPage) {
+  public interact(id, currentPage, extravalue?) {
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.service.ts:108]: variable name is not correct

**Review by HarishGangula — CHANGES_REQUESTED** [review]

**Pull request merged**